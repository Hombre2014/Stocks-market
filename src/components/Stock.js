import React from 'react';
import './stock.css';

const Stock = (stock) => {
  const {
    name, symbol, price, change, percentage,
  } = stock;
  const changeFloat = parseFloat(change).toFixed(2);
  const percentageFloat = parseFloat(percentage).toFixed(2);
  return (
    <div className="card">
      <h4>{name}</h4>
      <h5>
        Symbol:
        {' '}
        {symbol}
      </h5>
      <p>
        Price: $
        {price}
      </p>
      <p>
        Change:
        {' '}
        <span style={{ color: (changeFloat > 0) ? 'black' : 'red' }}>
          {changeFloat}
        </span>
      </p>
      <p>
        Percent change:
        {' '}
        <span style={{ color: (percentageFloat > 0) ? 'black' : 'red' }}>
          {percentageFloat}
          %
        </span>
      </p>
    </div>
  );
};

export default Stock;
