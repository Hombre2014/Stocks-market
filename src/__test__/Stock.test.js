import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Stock from '../components/Stock';
import store from '../redux/configureStore';

describe('Testing the stock data', () => {
  test('Renders stock data correctly', () => {
    const data = {
      symbol: 'AAPL',
      price: '174.05',
      change: -3.27,
      change_percentage: '-0.09726409',
      company_name: 'Apple Inc.',
    };
    const stock = render(
      <Provider store={store}>
        <BrowserRouter>
          <Stock data={data} />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(stock).toMatchSnapshot();
  });
});
