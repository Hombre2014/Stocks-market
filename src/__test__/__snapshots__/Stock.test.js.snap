// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the stock data Renders stock data correctly for ticker AAPL 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="stock-container"
      >
        <div
          class="stock-header"
        >
          <h4 />
          <h5>
            Symbol:
             
          </h5>
        </div>
        <div
          class="stock-details"
        >
          <p>
            Price: $
          </p>
          <p>
            Change: $
             
            <span
              style="color: rgb(255, 229, 229);"
            >
              NaN
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(134, 228, 141); display: none;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(255, 0, 0); display: none;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
              />
            </svg>
          </p>
          <p>
            Percent change:
             
            <span
              style="color: rgb(255, 229, 229);"
            >
              NaN
              %
            </span>
          </p>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="stock-container"
    >
      <div
        class="stock-header"
      >
        <h4 />
        <h5>
          Symbol:
           
        </h5>
      </div>
      <div
        class="stock-details"
      >
        <p>
          Price: $
        </p>
        <p>
          Change: $
           
          <span
            style="color: rgb(255, 229, 229);"
          >
            NaN
          </span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(134, 228, 141); display: none;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 0, 0); display: none;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
            />
          </svg>
        </p>
        <p>
          Percent change:
           
          <span
            style="color: rgb(255, 229, 229);"
          >
            NaN
            %
          </span>
        </p>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the stock data Renders stock data correctly for ticker F 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="stock-container"
      >
        <div
          class="stock-header"
        >
          <h4 />
          <h5>
            Symbol:
             
          </h5>
        </div>
        <div
          class="stock-details"
        >
          <p>
            Price: $
          </p>
          <p>
            Change: $
             
            <span
              style="color: rgb(255, 229, 229);"
            >
              NaN
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(134, 228, 141); display: none;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(255, 0, 0); display: none;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
              />
            </svg>
          </p>
          <p>
            Percent change:
             
            <span
              style="color: rgb(255, 229, 229);"
            >
              NaN
              %
            </span>
          </p>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="stock-container"
    >
      <div
        class="stock-header"
      >
        <h4 />
        <h5>
          Symbol:
           
        </h5>
      </div>
      <div
        class="stock-details"
      >
        <p>
          Price: $
        </p>
        <p>
          Change: $
           
          <span
            style="color: rgb(255, 229, 229);"
          >
            NaN
          </span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(134, 228, 141); display: none;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 0, 0); display: none;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
            />
          </svg>
        </p>
        <p>
          Percent change:
           
          <span
            style="color: rgb(255, 229, 229);"
          >
            NaN
            %
          </span>
        </p>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
